src/rsa.rs

```diff
@@ -163,8 +165,107 @@ fn parse_public_key<'a>(input: untrusted::Input<'a>) ->
     })
 }
 
+#[allow(non_camel_case_types)]
+enum BIGNUM {}
+#[allow(non_camel_case_types)]
+enum BN_MONT_CTX {}
+
+#[allow(non_camel_case_types)]
+#[repr(C)]
+struct RSA {
+    n: *mut BIGNUM,
+    e: *mut BIGNUM,
+    d: *mut BIGNUM,
+    p: *mut BIGNUM,
+    q: *mut BIGNUM,
+    dmp1: *mut BIGNUM,
+    dmq1: *mut BIGNUM,
+    iqmp: *mut BIGNUM,
+    mont_n: *mut BN_MONT_CTX,
+    mont_p: *mut BN_MONT_CTX,
+    mont_q: *mut BN_MONT_CTX,
+    mont_qq: *mut BN_MONT_CTX,
+    qmn_mont: *mut BIGNUM,
+    iqmp_mont: *mut BIGNUM,
+}
+
+/// An RSA key pair.
+pub struct RSAKeyPair {
+    rsa: Box<RSA>,
+}
+
+const BN_FLG_CONSTTIME: c::int = 4;
+
+fn build_bn(input: untrusted::Input) -> *mut BIGNUM {
+    unsafe {
+        let bytes = input.as_slice_less_safe();
+        let res = BN_bin2bn(bytes.as_ptr(), bytes.len(), ptr::null_mut());
+        BN_set_flags(res, BN_FLG_CONSTTIME);
```

Comment:
I was wrong in my other comment: We do need to call `BN_set_flags` to set the `BN_FLG_CONSTTIME` flag here. BUT, not for `n` and `e`, because those aren't private values.